lib/ddtrace/tracer.rb

```diff
@@ -164,8 +164,10 @@ def trace(name, options = {})
       span = Span.new(self, name, options)
 
       # set up inheritance
-      parent = @buffer.get()
-      span.set_parent(parent)
+      if span.parent_id.zero? || span.parent_id.nil?
+        parent = @buffer.get()
```

Comment:
Side note, this would probably need to be merged with https://github.com/DataDog/dd-trace-rb/pull/145 but I could take care of that (the change concerning this  `buffer` is not small, it... disappeared).